src/bentoml/_internal/utils/__init__.py

```diff
@@ -303,6 +304,24 @@ def resolve_user_filepath(filepath: str, ctx: t.Optional[str]) -> str:
     raise FileNotFoundError(f"file {filepath} not found")
 
 
+def find_nested_requirements_txts(
+    base_requirements_txt: Path,
+) -> t.Tuple[Path, t.Set[Path]]:
```

Comment:
```suggestion
) -> tuple[Path, set[Path]]:
```

Reply:
That's only valid syntax in Python 3.9+ isn't it?

Reply:
Type annotations aren't evaluated with `from __future__ import annotations`, so we use the new syntax for type annotations.